core/network/src/config.rs

```diff
@@ -372,7 +372,7 @@ pub enum NodeKeyConfig {
 pub type Ed25519Secret = Secret<ed25519::SecretKey>;
 
 /// The configuration options for obtaining a secret key `K`.
-#[derive(Clone)]
+#[derive(Clone, Debug)]
```

Comment:
I'd prefer a manual implementation of `Debug` here that doesn't print the value of `K`.

Something like:

```rust
impl<K> fmt::Debug for Secret<K> {
	fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
		match self {
			Secret::Input(_) => f.debug_tuple("Secret::Input").finish(),
			Secret::File(path) => f.debug_tuple("Secret::File").field(path).finish(),
			Secret::New => f.debug_tuple("Secret::New").finish(),
		}
	}
}
```

In practice I think that we never use anything like `Secret<Vec<u8>>`, but I think we're never careful enough when it comes to this.

Reply:
Thanks! Turns out that this suggestions is pretty good already so I simply copied and pasted